### DISCUSSION TITLE
How to pass the CI workflow/pipeline check

### DISCUSSION THREAD

**Discussion by ZuxingGit** [ZuxingGit/pygments#13] [Show and tell]
1. put a sample code file 1729.txt into path: tests/snippets/cpp
2. in command window, go into this folder: tests/snippets/cpp
3. run: tox -- --update-goldens *.txt
4. tox -- tests/snippets/cpp/ (must pass all)
5. go into folder: tests/examplefiles
6. run: tox -- --update-goldens
7. tox -- tests/examplefiles (must pass all)

**Now, the problem is: other lexers' test files were changed too, are they correct?**

reference:
https://pygments.org/docs/lexerdevelopment/#how-to-test-your-lexer
<img width="793" alt="image" src="https://github.com/ZuxingGit/pygments/assets/30425940/885d90f5-25b6-42db-b5b4-067664433388">

**Comment by adluu2k1** [comment]
Running the 6th command in tests/examplefiles will change all output from other lexers to the output from our code, making the 7th command useless (it will ALWAYS pass). I believe --update-goldens is only used in the lexer we believe to be affected, so 4 and 6 are not necessary. The correct procedure should be 1 (if we have new samples) -> 2 -> 3 -> 5 -> 7.

**Reply by ZuxingGit** [reply]
If I don't do 6. Then in 7, it will show these failed files:
<img width="1078" alt="image" src="https://github.com/ZuxingGit/pygments/assets/30425940/80970f91-b0d4-48e3-89d9-9ab06b3c270d">

**Comment by adluu2k1** [comment]
There seems to be outputs in the master branch in both repo (ours and original). I guess we could just pull these outputs. Or if you want to update everything with tox, do it in the master branch and bring outputs to the issuefix branch. Then, in issuefix branch, update only cpp outputs with --update-goldens tests/examplefiles/cpp. Then run tox without --update-goldens on all lexers (tests/examplefiles) to check outputs.

**Reply by ZuxingGit** [reply]
I revert those 13 test files, then run 2 commands as u said. Now only showing 10 failed files:
<img width="1153" alt="image" src="https://github.com/ZuxingGit/pygments/assets/30425940/c35bb600-9c33-4fad-8f50-14e68d713c55">
by comparison, 3 cpp test files disappeared.

**Comment by trungtruong1** [comment]
I think @adluu2k1 is right. We already have the ground truth from the `master` branch. One more thing, our change should not make the previous ground truth files incorrect. We should try to fix the lexer so that it can fix the issue while not tampering with the original ground truth files (I expect the repo owner to use original ground truth files to check if our change passes CI)

**Reply by ZuxingGit** [reply]
Should I revert those 13 files back to original version?
